doc/v5.0/man3/MPI_T_cvar_get_info.3
.\" -*- nroff -*-
.\" Copyright 2013 Los Alamos National Security, LLC. All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright (c) 2010 Cisco Systems, Inc.  All rights reserved.
.\" $COPYRIGHT$
.TH MPI_T_cvar_get_info 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.
.SH NAME
\fBMPI_T_cvar_get_info\fP \- Query information from a control variable
.
.SH SYNTAX
.ft R
.
.SH C Syntax
.nf
#include <mpi.h>
int MPI_T_cvar_get_info(int \fIcvar_index\fP, char *\fIname\fP, int *\fIname_len\fP,
                        int *\fIverbosity\fP, MPI_Datatype *\fIdatatype\fP, MPI_T_enum *\fIenumtype\fP,
                        const *\fIdesc\fP, int *\fIdesc_len\fP, int *\fIbind\fP, int *\fIscope\fP)

.fi
.SH INPUT PARAMETERS
.ft R
.TP 1i
cvar_index
Index of the control variable to be queried.
.
.
.SH INPUT/OUTPUT PARAMETERS
.ft R
.TP 1i
name_len
Length of the string and/or buffer for name.
.TP 1i
desc_len
Length of the string and/or buffer for desc.
.
.
.SH OUTPUT PARAMETERS
.ft R
.TP 1i
name
Buffer to return the string containing the name of the
control variable.
.TP 1i
verbosity
Verbosity level of this variable.
.TP 1i
datatype
MPI datatype of the information stored in the control
variable.
.TP 1i
enumtype
Optional descriptor for enumeration information.
.TP 1i
desc
Buffer to return the string containing the description
of the control variable.
.TP 1i
bind
Type of MPI object to which this variable must be
bound.
.TP 1i
scope
Scope of when changes to this variable are possible.
.
.
.SH DESCRIPTION
.ft R
MPI_T_cvar_get_info can be used to query information about a control variable. The function returns
the verbosity, datatype, enumeration type, binding, and scope of the queried control variable in the arguments
\fIverbosity\fP, \fIdatatype\fP, \fIenumtype\fP, \fIbind\fP, and \fIscope\fP, respectively. Control variables
in Open MPI are the same as MCA parameters.
.
.
.SH VERBOSITY
.ft R
As Open MPI exposes a very large number of MCA parameters (control variables), control variables are
categorized into nine verbosity levels corresponding to the equivalent ompi_info level. The nine levels are
(in increasing order):
.TP 1i
MPI_T_VERBOSITY_USER_BASIC
Basic information of interest to users
.TP 1i
MPI_T_VERBOSITY_USER_DETAIL
Detailed information of interest to users
.TP 1i
MPI_T_VERBOSITY_USER_ALL
All remaining information of interest to users
.TP 1i
MPI_T_VERBOSITY_TUNER_BASIC
Basic information required for tuning
.TP 1i
MPI_T_VERBOSITY_TUNER_DETAIL
Detailed information required for tuning
.TP 1i
MPI_T_VERBOSITY_TUNER_ALL
All remaining information required for tuning
.TP 1i
MPI_T_VERBOSITY_MPIDEV_BASIC
Basic information for MPI implementors
.TP 1i
MPI_T_VERBOSITY_MPIDEV_DETAIL
Detailed information for MPI implementors
.TP 1i
MPI_T_VERBOSITY_MPIDEV_ALL
All remaining information for MPI implementors

For more information see MPI-3 \[char167] 14.3.1.

.SH DATATYPE
.ft R
The datatype returned by MPI_T_cvar_get_info is restricted to one of the following datatypes: MPI_INT,
MPI_UNSIGNED, MPI_UNSIGNED_LONG, MPI_UNSIGNED_LONG_LONG, MPI_COUNT, MPI_CHAR, and MPI_DOUBLE. For more
information on datatypes in MPI_T see MPI-3 \[char167] 14.3.5.

.SH SCOPE
.ft R
The scope describes when and how changes can be made to a control variable. From MPI-3 \[char167] 14.3.6, the scope may be any of the following:
.TP 1i
MPI_T_SCOPE_CONSTANT
read-only, value is constant
.TP 1i
MPI_T_SCOPE_READONLY
read-only, cannot be written, but can change
.TP 1i
MPI_T_SCOPE_LOCAL
may be writeable, writing is a local operation
.TP 1i
MPI_T_SCOPE_GROUP
may be writeable, must be done to a group of processes, all processes in a group must be set to consistent values
.TP 1i
MPI_T_SCOPE_GROUP_EQ
may be writeable, must be done to a group of processes, all processes in a group must be set to the same value
.TP 1i
MPI_T_SCOPE_ALL
may be writeable, must be done to all processes, all connected processes must be set to consistent values
.TP 1i
MPI_T_SCOPE_ALL_EQ
may be writeable, must be done to all processes, all connected processes must be set to the same value

For more information see MPI-3 \[char167] 14.3.6 Table 14.4.

.SH NOTES
.ft R
This MPI tool interface function returns two strings. This function takes two argument for each string:
a buffer to store the string, and a length which must initially specify the size of the buffer. If the
length passed is n then this function will copy at most n - 1 characters of the string into the
corresponding buffer and set the length to the number of characters copied - 1. If the length argument
is NULL or the value specified in the length is 0 the corresponding string buffer is ignored and the
string is not returned.
.sp
Open MPI does not currently support binding control variables to MPI objects.
.
.
.SH ERRORS
.ft R
MPI_T_cvar_get_info() will fail if:
.TP 1i
[MPI_T_ERR_NOT_INITIALIZED]
The MPI Tools interface not initialized
.TP 1i
[MPI_T_ERR_INVALID_INDEX]
The control variable index is invalid
.
.SH SEE ALSO
.ft R
.nf
ompi_info


doc/v3.0/man3/MPI_T_cvar_get_info.3.php
<?php
$topdir = "../../..";
$title = "MPI_T_cvar_get_info(3) man page (version 3.0.6)";
$meta_desc = "Open MPI v3.0.6 man page: MPI_T_CVAR_GET_INFO(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
      <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_T_cvar_get_info</b> - Query information from a control variable

<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_T_cvar_get_info(int cvar_index, char *name, int *name_len,
                        int *verbosity, MPI_Datatype *datatype, MPI_T_enum
*enumtype,
                        const *desc, int *desc_len, int *bind, int *scope)
</pre>
<h2><a name='sect3' href='#toc3'>Input Parameters</a></h2>

<dl>

<dt>cvar_index </dt>
<dd>Index of the control variable to be queried.
  </dd>
</dl>

<h2><a name='sect4' href='#toc4'>Input/Output Parameters</a></h2>

<dl>

<dt>name_len </dt>
<dd>Length of the string and/or buffer for
name. </dd>

<dt>desc_len </dt>
<dd>Length of the string and/or buffer for desc.   </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameters</a></h2>

<dl>

<dt>name
</dt>
<dd>Buffer to return the string containing the name of the control variable.
</dd>

<dt>verbosity </dt>
<dd>Verbosity level of this variable. </dd>

<dt>datatype </dt>
<dd>MPI datatype of the
information stored in the control variable. </dd>

<dt>enumtype </dt>
<dd>Optional descriptor
for enumeration information. </dd>

<dt>desc </dt>
<dd>Buffer to return the string containing
the description of the control variable. </dd>

<dt>bind </dt>
<dd>Type of MPI object to which
this variable must be bound. </dd>

<dt>scope </dt>
<dd>Scope of when changes to this variable
are possible.   </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
MPI_T_cvar_get_info can be used to query information
about a control variable. The function returns the verbosity, datatype,
enumeration type, binding, and scope of the queried control variable in
the arguments <i>verbosity</i>, <i>datatype</i>, <i>enumtype</i>, <i>bind</i>, and <i>scope</i>, respectively.
Control variables in Open MPI are the same as MCA parameters.
<h2><a name='sect7' href='#toc7'>Verbosity</a></h2>
As
Open MPI exposes a very large number of MCA parameters (control variables),
control variables are categorized into nine verbosity levels corresponding
to the equivalent ompi_info level. The nine levels are (in increasing order):

<dl>

<dt>MPI_T_VERBOSITY_USER_BASIC </dt>
<dd>Basic information of interest to users </dd>

<dt>MPI_T_VERBOSITY_USER_DETAIL
</dt>
<dd>Detailed information of interest to users </dd>

<dt>MPI_T_VERBOSITY_USER_ALL </dt>
<dd>All
remaining information of interest to users </dd>

<dt>MPI_T_VERBOSITY_TUNER_BASIC
</dt>
<dd>Basic information required for tuning </dd>

<dt>MPI_T_VERBOSITY_TUNER_DETAIL </dt>
<dd>Detailed
information required for tuning </dd>

<dt>MPI_T_VERBOSITY_TUNER_ALL </dt>
<dd>All remaining
information required for tuning </dd>

<dt>MPI_T_VERBOSITY_MPIDEV_BASIC </dt>
<dd>Basic information
for MPI implementors </dd>

<dt>MPI_T_VERBOSITY_MPIDEV_DETAIL </dt>
<dd>Detailed information
for MPI implementors </dd>

<dt>MPI_T_VERBOSITY_MPIDEV_ALL </dt>
<dd>All remaining information

<p>for MPI implementors
<p> For more information see MPI-3 [char167] 14.3.1.
<p> </dd>
</dl>

<h2><a name='sect8' href='#toc8'>Datatype</a></h2>
The
datatype returned by MPI_T_cvar_get_info is restricted to one of the following
datatypes: MPI_INT, MPI_UNSIGNED, MPI_UNSIGNED_LONG, MPI_UNSIGNED_LONG_LONG,
MPI_COUNT, MPI_CHAR, and MPI_DOUBLE. For more information on datatypes in
MPI_T see MPI-3 [char167] 14.3.5.
<p>
<h2><a name='sect9' href='#toc9'>Scope</a></h2>
The scope describes when and how changes
can be made to a control variable. From MPI-3 [char167] 14.3.6, the scope may
be any of the following:
<dl>

<dt>MPI_T_SCOPE_CONSTANT </dt>
<dd>read-only, value is constant
</dd>

<dt>MPI_T_SCOPE_READONLY </dt>
<dd>read-only, cannot be written, but can change </dd>

<dt>MPI_T_SCOPE_LOCAL
</dt>
<dd>may be writeable, writing is a local operation </dd>

<dt>MPI_T_SCOPE_GROUP </dt>
<dd>may be
writeable, must be done to a group of processes, all processes in a group
must be set to consistent values </dd>

<dt>MPI_T_SCOPE_GROUP_EQ </dt>
<dd>may be writeable,
must be done to a group of processes, all processes in a group must be
set to the same value </dd>

<dt>MPI_T_SCOPE_ALL </dt>
<dd>may be writeable, must be done to
all processes, all connected processes must be set to consistent values
</dd>

<dt>MPI_T_SCOPE_ALL_EQ </dt>
<dd>may be writeable, must be done to all processes, all

<p>connected processes must be set to the same value
<p> For more information
see MPI-3 [char167] 14.3.6 Table 14.4.
<p> </dd>
</dl>

<h2><a name='sect10' href='#toc10'>Notes</a></h2>
This MPI tool interface function
returns two strings. This function takes two argument for each string: a
buffer to store the string, and a length which must initially specify the
size of the buffer. If the length passed is n then this function will copy
at most n - 1 characters of the string into the corresponding buffer and
set the length to the number of characters copied - 1. If the length argument
is NULL or the value specified in the length is 0 the corresponding string
buffer is ignored and the string is not returned. <p>
Open MPI does not currently
support binding control variables to MPI objects.
<h2><a name='sect11' href='#toc11'>Errors</a></h2>
MPI_T_cvar_get_info()
will fail if:
<dl>

<dt>[MPI_T_ERR_NOT_INITIALIZED] </dt>
<dd>The MPI Tools interface not initialized
</dd>

<dt>[MPI_T_ERR_INVALID_INDEX] </dt>
<dd>The control variable index is invalid  </dd>
</dl>

<h2><a name='sect12' href='#toc12'>See Also</a></h2>
<br>
<pre>ompi_info

<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
